versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5771 @@
+The `command` section is the only required task section. It defines the command template that is evaluated and executed when the task is called. Specifically, the commands are executed after all of the inputs are staged and before the outputs are evaluated.
+
+There are two different syntaxes that can be used to define the command section:
+
+```wdl
+# HEREDOC style - this way is preferred
+command <<< ... >>>
+
+# older style - may be preferable in some cases
+command { ... }
+```
+
+There may be any number of commands within a command section. Commands are not modified by the execution engine, with the following exceptions:
+- Common leading whitespace is [stripped](#stripping-leading-whitespace)
+- String interpolation is performed on the entire command section to replace [expression placeholders](#expression-placeholders) with their actual values.
+
+#### Expression Placeholders
+
+The body of the command section (i.e. the command "template") can be though of as a single string expression, which (like all string expressions) may contain placeholders.
+
+There are two different syntaxes that can be used to define command expression placeholders, depending on which style of command section definition is used:
+
+|Command Definition Style|Placeholder Style|
+|---|---|
+|`command { ... }`|`~{}` (preferred) or `${}`|
+|`command <<< >>>`|`~{}` only|
+
+Note that the `~{}` style of placeholder may only be used within the `command` section - for all other string expressions, the `${}` style must be used.
+
+Any valid WDL expression may be used within a placeholder. For example, a command might reference an input to the task, like this:
+
+```wdl
+task test {
+  input {
+    File infile
+  }
+  command <<<
+    cat ~{infile}
+  >>>
+  ....
+}
+```
+
+In this case, `infile` within the `~{...}` is an identifier expression referencing the value of the `infile` input parameter that was specified at runtime. Since `infile` is a `File` declaration, the execution engine will have staged whatever file was referenced by the caller such that it is available on the local file system, and will have replaced the original value of the `infile` parameter with the path to the file on the local filesystem.
+
+The expression can also be more complex, for example a function call: 
+
+```wdl
+task write_array {
+  input {
+    Array[String] str_array
+  }
+  command <<<
+    # the "write_lines" function writes each string in an array
+    # as a line in a temporary file, and returns the path to that
+    # file, which can then be referenced by other commands such as 
+    # the unix "cat" command
+    cat write_lines(str_array)
+  >>>
+}
+```
+
+In most cases, the `~{}` style of placeholder is preferred, to avoid ambiguity between WDL placeholders and Bash variables, which are of the form `$name` or `${name}`. If the `command { ... }` style is used, then `${name}` is always interpreted as a WDL placeholder, so care must be taken to only use `$name` style Bash variables.
```

Comment:
At the end of the paragraph "Or use `command <<< ... >>>` style". It is good to make that explicit.

Reply:
done